uploader/push_to_sling.py

```diff
@@ -36,74 +68,114 @@
 if pw == '-':
     pw = getpass.getpass()
 
-print('Using server: ' + args.server)
-print('Using repository: ' + args.repository)
-print('--------------')
+config = None
+try:
+    config = yaml.safe_load(open(args.directory + '/pantheon2.yml'))
+except FileNotFoundError:
+    logger.warning('Could not find a valid config file in this directory; all files will be treated as resource uploads.')
```

Comment:
Would a better default be perhaps to stop the upload altogether if no config file is found?

Reply:
I thought about that and I could be convinced that that's the way to go, but if we're going to be requiring the config file anyway, then we may as well drop half of the command line arguments. (Maybe that's a good thing for simplicity's sake?)

I had thought that maybe some users would want to script their uploads and perhaps having the options exposed as arguments rather than a file-based config might be more convenient for that, but maybe that's not the case.

Reply:
that makes sense. so maybe what we need to make sure is that all parameters can be provided either via config file *or* command line, and which one precedes the other.

Reply:
My thought for priority was: command line > config file > hardcoded default - for all options.

Technically we have the ability to specify arguments more than once and capture them in an array (I really like this library!), so we COULD let users specify the globs on the command line as well. That seems like gold-plating though - what do you think?

Reply:
makes sense to me :+1: